src/strat_engine/blockdev.rs

```diff
@@ -55,87 +55,109 @@ impl BlockDev {
                mda_size: Sectors,
                force: bool)
                -> EngineResult<BlockDev> {
-        let metadata = try!(fs::metadata(path));
+        Ok(try!(Self::batch_new(stratis_id, &[path], mda_size, force)).pop().unwrap())
+    }
 
-        if !metadata.file_type().is_block_device() {
-            return Err(EngineError::Io(io::Error::new(ErrorKind::InvalidInput,
-                                                      format!("{} is not a block device",
-                                                              path.display()))));
-        }
+    /// Initialize multiple blockdevs at once. This allows all of them
+    /// to be checked for usability before writing to any of them.
+    pub fn batch_new(stratis_id: &str,
+                     paths: &[&Path],
+                     mda_size: Sectors,
+                     force: bool)
+                     -> EngineResult<Vec<BlockDev>> {
 
         if *mda_size % 2 != 0 {
             return Err(EngineError::Io(io::Error::new(ErrorKind::InvalidInput,
-                                                      format!("{} mda size is not an even \
+                                                      format!("mda size {} is not an even \
                                                                number",
-                                                              path.display()))));
+                                                              *mda_size))));
         }
 
         if mda_size < MIN_MDA_SIZE {
             return Err(EngineError::Io(io::Error::new(ErrorKind::InvalidInput,
-                                                      format!("{} mda size is less than \
+                                                      format!("mda size {} is less than \
                                                                minimum ({})",
-                                                              path.display(),
+                                                              *mda_size,
                                                               *MIN_MDA_SIZE))));
         }
 
-        let dev = try!(Device::from_str(&path.to_string_lossy()));
+        // Check stuff for all candidates before writing anything
+        let mut good_paths = Vec::new();
```

Comment:
The goal was really to hold on to dev and dev_size values that we needed to get (to check if they're valid) but also we need to initialize fields in BlockDev, once all tests are passed. Paths doesn't need to be in there -- we can just zip the `vec<(dev, dev_size)>` with the `vec<paths>` in the following loop, and call the former "path_info" perhaps?